drivers/uktty/ns16550/ns16550.c

```diff
@@ -69,11 +88,15 @@
 static __u8 ns16550_uart_initialized;
 static __u64 ns16550_uart_base;
 
+#if CONFIG_LIBUKLIBPARAM
+UK_LIBPARAM_PARAM_ALIAS(base, &ns16550_uart_base, __u64, "ns15550 base");
```

Comment:
Is here a typo in the help? `ns16550 base`?
Btw, this PR removes the capability to use the driver without fdt and without kernel command line parser. Wouldn't it be useful to also have the option to configure defaults for that scenario?

Reply:
> Is here a typo in the help? ns16550 base?

Of course 👍🏼 

> Btw, this PR removes the capability to use the driver without fdt and without kernel command line parser. Wouldn't it be useful to also have the option to configure defaults for that scenario?

That's right. Supporting boot without fdt would require that we provide Kconfig options for all drivers and for all their config options, the amount of config options introduced and their impact in the code would be pretty high. IMO this is not worth to support as the device-tree is the standard (and only) way to boot on arm-based embedded systems, that this requirement shouldn't come as a surprise. So unless we have a real use-case in mind, I would vouch for keeping things simple. What do you think?

Reply:
I am more thinking towards virtualized environments, especially smaller VMMs that are ported from the x86 world and in particular I am thinking of firecracker. Does it use fdt's for booting? If so, then it is fine to take that as assumption for Arm.

Reply:
Yes, FC on arm64 also passes a device-tree blob it generates, and similarly for `kvm-tool` I am not aware of a VMM that doesn't provide an fdt for arm64.

Reply:
Okay, fair enough. Makes sense. :+1: